### PR TITLE
Add Kotlin tabs to Extension Model topics

### PR DESCRIPTION
**Pull request by danil-pavlov**
Adds Kotlin tabbed code snippets to the Extension Model section of the User Guide.

Pages updated:

* Exception Handling
* Intercepting Invocations
* Providing Invocation Contexts for Class Templates
* Providing Invocation Contexts for Test Templates
* Relative Execution Order of User Code and Extensions

Plus, a possible error in the `DatabaseTestsDemo` Java test class and its output was fixed.

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit-framework/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [ ] There are no TODOs left in the code
- [ ] Method [preconditions](https://docs.junit.org/snapshot/api/org.junit.platform.commons/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [ ] [Coding conventions](https://github.com/junit-team/junit-framework/blob/HEAD/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [ ] Change is covered by [automated tests](https://github.com/junit-team/junit-framework/blob/HEAD/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [ ] Public API has [Javadoc](https://github.com/junit-team/junit-framework/blob/HEAD/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [ ] Change is documented in the [User Guide](https://docs.junit.org/snapshot/) and [Release Notes](https://docs.junit.org/snapshot/release-notes.html)

### CODE REVIEW

File: documentation/modules/ROOT/pages/extensions/exception-handling.adoc [hunk 1]
```diff
@@ -10,27 +10,59 @@ and for those thrown during one of test lifecycle methods (`@BeforeAll`, `@Befor
...
+[source,kotlin,indent=0]
+----
+include::example$kotlin/example/kotlin/exception/IgnoreIOExceptionExtension.kt[tags=user_guide]
+----
+--
+====
 
 Another example shows how to record the state of an application under test exactly at
-the point of unexpected exception being thrown during setup and cleanup. Note that unlike
+the point of an unexpected exception being thrown during setup and cleanup. Note that unlike
```
Comment (mpkorstanje) [inline documentation/modules/ROOT/pages/extensions/exception-handling.adoc]:
```suggestion
the point where an unexpected exception is thrown during setup and cleanup. Note that unlike
```

File: documentation/modules/ROOT/pages/extensions/providing-invocation-contexts-for-class-templates.adoc [hunk 2]
```diff
@@ -33,9 +49,9 @@ output when using the `ConsoleLauncher` is as follows.
 
 The `{ClassTemplateInvocationContextProvider}` extension API is primarily intended for
 implementing different kinds of tests that rely on repetitive invocation of _all_ test
-methods in a test class albeit in different contexts — for example, with different
+methods in a test class, albeit in different contexts — for example, with different
 parameters, by preparing the test class instance differently, or multiple times without
 modifying the context.
 Please refer to the implementations of
```
Comment (mpkorstanje) [inline documentation/modules/ROOT/pages/extensions/providing-invocation-contexts-for-class-templates.adoc]:
```suggestion
Please refer to the implementation of
```

File: documentation/modules/ROOT/pages/extensions/providing-invocation-contexts-for-class-templates.adoc [hunk 3]
```diff
@@ -33,9 +49,9 @@ output when using the `ConsoleLauncher` is as follows.
...
-xref:writing-tests/parameterized-classes-and-tests.adoc[Parameterized Classes] which uses this extension
+xref:writing-tests/parameterized-classes-and-tests.adoc[Parameterized Classes] which use this extension
```
Comment (mpkorstanje) [inline documentation/modules/ROOT/pages/extensions/providing-invocation-contexts-for-class-templates.adoc]:
```suggestion
xref:writing-tests/parameterized-classes-and-tests.adoc[Parameterized Classes] which uses this extension
```

There is only one implementation of `@ParameterizedClass`.

**Review by mpkorstanje — COMMENTED** [review]
LGTM. Some language nitpicks, but I'm not 100% sure about any of them.

File: documentation/src/test/kotlin/example/kotlin/ClassTemplateDemo.kt [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+    @Test
+    fun wellKnown() {
+        assertTrue(fruit in WELL_KNOWN_FRUITS)
+    }
+
+    class MyClassTemplateInvocationContextProvider : ClassTemplateInvocationContextProvider {
+        override fun supportsClassTemplate(context: ExtensionContext) = true
+
+        override fun provideClassTemplateInvocationContexts(context: ExtensionContext): Stream<ClassTemplateInvocationContext> =
```
Comment (marcphilipp) [inline documentation/src/test/kotlin/example/kotlin/ClassTemplateDemo.kt:44]: Can the explicit return type declaration be omitted?

```suggestion
        override fun provideClassTemplateInvocationContexts(context: ExtensionContext) =
```

File: documentation/src/test/kotlin/example/kotlin/TestTemplateDemo.kt [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    @TestTemplate
+    @ExtendWith(MyTestTemplateInvocationContextProvider::class)
+    fun testTemplate(fruit: String) {
+        assertTrue(fruit in fruits)
+    }
+
+    class MyTestTemplateInvocationContextProvider : TestTemplateInvocationContextProvider {
+        override fun supportsTestTemplate(context: ExtensionContext) = true
+
+        override fun provideTestTemplateInvocationContexts(context: ExtensionContext): Stream<TestTemplateInvocationContext> =
```
Comment (marcphilipp) [inline documentation/src/test/kotlin/example/kotlin/TestTemplateDemo.kt:37]: Ditto

```suggestion
        override fun provideTestTemplateInvocationContexts(context: ExtensionContext) =
```

**Review by marcphilipp — APPROVED** [review]

**Review by mpkorstanje — APPROVED** [review]
